ui/js/dfv/src/fields/datetime/index.js

```diff
@@ -119,6 +119,9 @@ const DateTime = ( {
 
 	const useHTML5Field = toBool( html5 ) && checkForHTML5BrowserSupport( 'datetime-local' );
 
+	const allowEmpty = toBool( fieldConfig.datetime_allow_empty ?? 0 );
+	const emptyValues = [ '0000-00-00', '0000-00-00 00:00:00', '00:00:00', '' ];
```

Comment:
The exception here is that 00:00:00 is a valid non-empty time right?

Reply:
Check, I can also move the valid empty values to the appropriate classes and let the `date` and `time` classes override the `datetime` main class.

Reply:
00:00:00 is an intentionally supported value and is treated as non-empty in other areas. I've removed it from the list here.